### PR TITLE
feat(Sagemaker): allow deploying llama 3.1 from Sagemaker Jumpstart

### PR DESCRIPTION
**Pull request by arief-hidayat**
Feature # allow deploying llama 3.1 from Sagemaker Jumpstart
---
1. add Jumpstart support for regions: me-central-1, ap-northeast-3, ap-southeast-3, il-central-1, us-gov-east-1, us-gov-west-1
2. add support for model with `gated_bucket` enabled such as llama 3.1
3. 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

File: src/common/helpers/kendra-helper.ts
Comment (krokoko) [inline src/common/helpers/kendra-helper.ts:1]: This change seems unrelated to this PR ?

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts [hunk 1]
```diff
@@ -183,6 +183,10 @@ export class JumpStartSageMakerEndpoint extends SageMakerEndpointBase {
       ...this.environment,
     };
 
+    if (environment.SAGEMAKER_SUBMIT_DIRECTORY) {
+      delete environment.SAGEMAKER_SUBMIT_DIRECTORY;
```
Comment (krokoko) [inline src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts:187]: any reason to delete this variable ?

**Review by krokoko — CHANGES_REQUESTED** [review]
Thanks for the PR ! Left a few comments
Also could you please fix the conflicts so we can run all the workflows

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts [hunk 2]
```diff
@@ -195,7 +199,8 @@ export class JumpStartSageMakerEndpoint extends SageMakerEndpointBase {
     vpcConfig: sagemaker.CfnModel.VpcConfigProperty | undefined,
   ) {
     const key = this.spec.prepackedArtifactKey ?? this.spec.artifactKey;
-    const bucket = JumpStartConstants.JUMPSTART_LAUNCHED_REGIONS[this.region]?.contentBucket;
+    const bucket = this.spec.gatedBucket ? JumpStartConstants.JUMPSTART_LAUNCHED_REGIONS[this.region]?.gatedContentBucket :
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts:202]: Please help me understand if adding the `gatedBucket` boolean will first be pulled.

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts (hunk 1 above)
Comment (arief-hidayat) [reply src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts:187]: based on my understanding this env var is for model training.
and somehow, when we have this env var, the Jumpstart UI console complaints that it is not detected as jumpstart model, hence the option to `Test Inference` is disabled.
after I removed this env var, now I can do manual test via `Test Inference` via console.

File: src/common/helpers/kendra-helper.ts
Comment (arief-hidayat) [reply src/common/helpers/kendra-helper.ts:1]: yes. this is more general. after several times deploying, the CDK failed with the following error.
```
Invalid Logging Configuration: The CloudWatch Logs Resource Policy size was exceeded. We suggest prefixing your CloudWatch lo
g group name with /aws/vendedlogs/states/.
```
I can exclude it from the PR it you prefer

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts (hunk 2 above)
Comment (arief-hidayat) [reply src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts:202]: apologise, I don't understand your statement.
if gatedBucket is true, we will use `gatedContentBucket` otherwise use `contentBucket`.
At least that's my observation during deploying llama, please feel free to correct me.

Comment (scottschreckengaust) [reply src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-sagemaker-endpoint.ts:202]: > apologise, I don't understand your statement. if gatedBucket is true, we will use `gatedContentBucket` otherwise use `contentBucket`. At least that's my observation during deploying llama, please feel free to correct me.

I miss interpreted that each model spec determines whether or not the `gatedBucket` exists or not.

**Review by krokoko — APPROVED** [review]

**Review by scottschreckengaust — APPROVED** [review]
LGTM
